lib/beyond_api/error.rb

```diff
@@ -5,6 +5,8 @@ class Error < StandardError
     attr_reader :error_id, :details, :trace_id, :full_message, :status_code, :error, :error_description
 
     def initialize(data, status_code = nil)
+      data ||= {}
+
```

Comment:
What about putting the default value on the `initialize`? (`def initialize(data = {}, status_code = nil)`)

Reply:
That works to make te parameter optional, but not to prevent `nil` value sent. 
E.g.

```
a_nil_value = nil
Error.new(a_nil_value)
```
`data` will be `nil` enven if we use `def initialize(data = {}, status_code = nil)`